arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/FindClosestFacilityInteractiveViewController.swift

```diff
@@ -0,0 +1,125 @@
+    
+    // Create an array of facilities in the area.
+    private var facilities = [AGSFacility]()
+    
+    // Add the facilities and create graphics.
+    private func createFacilitiesAndGraphics() {
+        facilities = [
+            AGSFacility(point: AGSPoint(x: -1.3042129900625112E7, y: 3860127.9479775648, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3042193400557665E7, y: 3862448.873041752, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3046882875518233E7, y: 3862704.9896770366, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3040539754780494E7, y: 3862924.5938606677, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3042571225655518E7, y: 3858981.773018156, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3039784633928463E7, y: 3856692.5980474586, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3049023883956768E7, y: 3861993.789732541, spatialReference: .webMercator()))]
+        let facilitySymbol = AGSPictureMarkerSymbol(url: facilityURL)
+        facilitySymbol.height = 30
+        facilitySymbol.width = 30
+        
+        for eachFacility in facilities {
+            facilityGraphicsOverlay.graphics.add(AGSGraphic(geometry: eachFacility.geometry, symbol: facilitySymbol, attributes: .none))
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["FindClosestFacilityInteractiveViewController"]
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+extension FindClosestFacilityInteractiveViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Find the closest facilities with the default parameters.
+        closestFacilityTask.defaultClosestFacilityParameters { [weak self] (parameters, error) in
+            guard let self = self else { return }
+            if let parameters = parameters {
+                parameters.setFacilities(self.facilities)
+                
+                // Create a symbol for the incident.
+                let incidentSymbol = AGSSimpleMarkerSymbol(style: .cross, color: .black, size: 20)
+                
+                // Remove previous graphics.
+                self.incidentGraphicsOverlay.graphics.removeAllObjects()
+                
+                // Create a point and graphics for the incident.
+                let incidentPoint = AGSPoint(x: mapPoint.x, y: mapPoint.y, spatialReference: .webMercator())
+                let graphic = AGSGraphic(geometry: incidentPoint, symbol: incidentSymbol, attributes: .none)
```

Comment:
From the documentation for `Optional.none`:

> In code, the absence of a value is typically written using the `nil` literal rather than the explicit `.none` enumeration case.

```swift
let graphic = AGSGraphic(geometry: incidentPoint, symbol: incidentSymbol, attributes: nil)
```

But since `attributes` has a default value of `nil`, the parameter can just be omitted:

```suggestion
                let graphic = AGSGraphic(geometry: incidentPoint, symbol: incidentSymbol)
```